File: LeetCode-Python/38-报数.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        else:
            st = self.countAndSay(n-1)
            count = 1
            res = ""
            for i in range(0, len(st)):
                if i+1 < len(st) and st[i] == st[i+1]:
                    count += 1
                else:
                    res = res+ str(count)+str(st[i])
                    count = 1 
        return res

File: LeetCode-Go/38-报数.go
func countAndSay(n int) string {
	if n == 1 {
		return "1"
	} else {
		str := countAndSay(n-1)
		count := 1
		res := ""
		for i:=0;i<len(str);i++ {
			if i+1<len(str)&&str[i]==str[i+1]{
				count ++
			} else {
			   res = res + strconv.Itoa(count)+string(str[i])
			   // 重置
			   count = 1
			}
		}
		return res
	}
}